src/main/java/seedu/address/logic/commands/AttendanceCommand.java

```diff
@@ -0,0 +1,215 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ATTENDANCE_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DROP_OFF;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PICK_UP;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Scanner;
+
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.pet.Name;
+import seedu.address.model.pet.Pet;
+import seedu.address.model.pet.Phone;
+import seedu.address.storage.JsonAdaptedAttendance;
+
+public class AttendanceCommand extends Command {
+
+    public static final String COMMAND_WORD = "attendance";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds an attendance of a pet to the database "
+            + "by the index number used in the displayed pet list. \n"
+            + "Includes the date, pick up time and drop off time (if any). \n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_ATTENDANCE_DATE + "DATE OF ATTENDANCE IN DD/MM/YYYY] "
+            + "[" + PREFIX_PICK_UP + "PICK UP TIME IN HH:MM] "
+            + "[" + PREFIX_DROP_OFF + "DROP OFF TIME IN HH:MM] \n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_ATTENDANCE_DATE + "15/03/2022 "
+            + PREFIX_PICK_UP + "09:15 "
+            + PREFIX_DROP_OFF + "18:00";
+
+    public static final String MESSAGE_ATTENDANCE_SUCCESS = "Attendance updated: %1$s";
+    public static final String MESSAGE_PET_NOT_ADDED = "Seems like this pet is not in the address book yet! "
+                                                        + "Make sure to add the pet in first!";
```

Comment:
`MESSAGE_PET_NOT_ADDED` appears to be confusing. Since the command uses `INDEX` to identify the pet, I believe if an invalid index is entered,  `MESSAGE_INVALID_PET_DISPLAYED_INDEX` will shown instead. I would recommend to rename this to  `MESSAGE_PET_FILE_ERROR` instead and amend message accordingly.